plugins/org.eclipse.fordiac.ide.typemanagement/src/org/eclipse/fordiac/ide/typemanagement/refactoring/delete/UpdateUntypedSubappPinChange.java

```diff
@@ -91,8 +91,8 @@ protected Command createCommand(final VarDeclaration varDecl) {
 	public RefactoringStatus isValid(final VarDeclaration element, final IProgressMonitor pm)
 			throws CoreException, OperationCanceledException {
 		final RefactoringStatus status = new RefactoringStatus();
-		if (!(element.eContainer() instanceof UntypedSubApp)) {
-			status.addFatalError(Messages.UpdateUntypedSubappPinChange_0);
+		if (!(element.eContainer() != null && element.eContainer().eContainer() instanceof UntypedSubApp)) {
```

Comment:
easier to read:
`element.eContainer() == null || !element.eContainer().eContainer() instanceof UntypedSubApp`

Reply:
even better !element.getFBNetworkElement() instanceof UntypedSubApp`